prism/src/main/java/org/apache/falcon/service/BacklogMetricEmitterService.java

```diff
@@ -0,0 +1,355 @@
+                    String metricName;
+                    Long backlog = backLogsCluster.get(clusterName) / (60 * 1000); // Converting to minutes
+                    if (pipelinesStr != null && !pipelinesStr.isEmpty()) {
+                        String[] pipelines = pipelinesStr.split(",");
+                        for (String pipeline : pipelines) {
+                            metricName = METRIC_PREFIX + METRIC_SEPARATOR + clusterName + METRIC_SEPARATOR
+                                    + pipeline + METRIC_SEPARATOR + LifeCycle.EXECUTION.name()
+                                    + METRIC_SEPARATOR + entityObj.getName() + METRIC_SEPARATOR
+                                    + "backlogInMins";
+                            metricNotificationService.publish(metricName, backlog);
+                        }
+                    } else {
+                        metricName = METRIC_PREFIX + METRIC_SEPARATOR + clusterName + METRIC_SEPARATOR
+                                + DEFAULT_PIPELINE + METRIC_SEPARATOR + LifeCycle.EXECUTION.name()
+                                + METRIC_SEPARATOR + entityObj.getName() + METRIC_SEPARATOR
+                                + "backlogInMins";
+                        metricNotificationService.publish(metricName, backlog);
+                    }
+                }
+            }
+        }
+    }
+
+
+    /**
+     * Service runs periodically and removes succeeded instances from backlog list.
+     */
+    public static class BacklogCheckService implements Runnable {
+
+        @Override
+        public void run() {
+            LOG.debug("BacklogCheckService running for entities");
+            try {
+                AbstractWorkflowEngine wfEngine = getWorkflowEngine();
+                for (Entity entity : entityBacklogs.keySet()) {
+                    List<MetricInfo> metrics = entityBacklogs.get(entity);
+                    if (!metrics.isEmpty()) {
+                        synchronized (metrics) {
+                            Iterator iterator = metrics.iterator();
+                            while (iterator.hasNext()) {
+                                MetricInfo metricInfo = (MetricInfo) iterator.next();
+                                String nominalTimeStr = metricInfo.getNominalTime();
+                                Date nominalTime;
+                                try {
+                                    nominalTime = DATE_FORMAT.get().parse(nominalTimeStr);
+                                    if (entity.getACL().getOwner() != null && !entity.getACL().getOwner().isEmpty()) {
+                                        CurrentUser.authenticate(entity.getACL().getOwner());
+                                    } else {
+                                        CurrentUser.authenticate(System.getProperty("user.name"));
+                                    }
+                                    if (wfEngine.isMissing(entity)) {
```

Comment:
I tried doing that but i found few problems in that approach, Once we have found entity was deleted , we have to store that entity in a list and check whether instances are coming or not. Until what time we have to maintain the list ? 
What if falcon restart happened after that entity was deleted and list will be cleared and we got instances from that entity saying instance missed sla high ?